src/crimson/admin/admin_socket.cc

```diff
@@ -0,0 +1,896 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2011 New Dream Network
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#include "common/version.h"
+#include "crimson/net/Socket.h"
+// activate for some inline unit-testing: #define UNIT_TEST_OSD_ASOK
+#include "crimson/admin/admin_socket.h"
+#include "crimson/common/log.h"
+//#include "seastar/testing/test_case.hh"
+#include "seastar/net/api.hh"
+#include "seastar/net/inet_address.hh"
+#include "seastar/core/reactor.hh"
+#include "seastar/core/thread.hh"
+//#include "seastar/util/log.hh"
+#include "seastar/util/std-compat.hh"
+#include <boost/algorithm/string.hpp>
+
+/*!
+  A Crimson-wise version of the admin socket - implementation file
+
+  \todo handle the unlinking of the admin socket. Note that 'cleanup_files' at-exit functionality is not yet
+        implemented in Crimson
+*/
+
+namespace {
+  seastar::logger& logger() {
+    return ceph::get_logger(ceph_subsys_osd);
+  }
+}
+
+#ifdef UNIT_TEST_OSD_ASOK
+namespace asok_unit_testing {
+  seastar::future<> utest_run_1(AdminSocket* asok);
+}
+#endif
+
+/*!
+  Hooks table - iterator support
+
+  Note: Each server-block is locked (via its gate member) before
+        iterating over its entries.
+ */
+
+AdminHooksIter AdminSocket::begin() {
+  AdminHooksIter it{*this};
+  it.m_miter->second.m_gate.enter();
+  it.m_in_gate = true;
+  return it;
+}
+
+AdminHooksIter AdminSocket::end() {
+  AdminHooksIter it{*this, true};
+  return it;
+}
+
+/*!
```

Comment:
Kind of a nit, but this doesn't really match any of the comment conventions used anywhere else in the code base (except apparently for crush.h and a couple of files in src/erasure-code).  Despite what CodingStyle says, our function/class comments tend to do the

/**
 *
 */

thing with occasionally with javadoc/kernel flavored tags for args/return/etc if that sort of thing seems useful.

Reply:
well - that's what seastar, SPDK, and some more libraries in our code use.
Doxygen supports all three options ( ///, /**, /*! ), and I followed what I use in Seastar changes.

Reply:
i think we are not obliged to follow any of the coding styles of seastar, SPDK or 3rd party libraries . let me put this way, what if they don't use the same coding styles? which one shall we follow?

Reply:
Yeah, let's stick with the
/**
 *
 */
style for uniformity.

Reply:
no problem